server/api/messages_api.js

```diff
@@ -1,152 +1,160 @@
-    if (!message) {
-      throw { status: 404 };
-    }
-    // A message may be viewed only by the sender, the receiver, or an admin.
-    if (!req.isAdmin && req.user.id != message.fromUserId && req.user.id != message.toUserId) {
-      throw { status: 401 };
-    }
-    return message;
-  }));
-});
-
-// Delete message by ID
-router.delete("/:id", function(req, res) {
-  res.jsonResultOf(Message.findById(req.params.id)
-  .then(function(message) {
-    if (!message) {
-      throw { status: 404 };
-    }
-    // A message may be deleted only by the sender.
-    if (!req.isAdmin && req.user.id != message.fromUserId) {
-      throw { status: 401 };
-    }
-    return Message.destroyById(req.params.id);
-  }));
-});
-
-if (process.env.NODE_ENV == "test") {
-  // Delete all messages
-  router.delete("/", function(req, res) {
-    res.jsonResultOf(new Promise(function(resolve) {
-      resolve(Message.destroyAll());
-    }))
-  });
-}
-
-// Create a message.
-router.post("/", function(req, res) {
-  res.jsonResultOf(new Promise(function(resolve) {
-    if (!req.user) {
-      throw { status: 401 };
-    }
-
-    var fields = VALIDATOR.validateNew(req.body);
-    fields.fromUserId = req.user.id;
-    
-    // TODO: validate asset.
-
-    // Validate toUserId.
-    var toUserId = fields.toUserId;
-    var personal = false;
-    switch (fields.type) {
-    case Message.GREETING_TYPE:
-    case Message.INVITE_TYPE:
-      personal = true;
-    }
-    if ((toUserId == null) == personal) {
-      throw { body: { toUserId: toUserId || "?" }};
-    }
-
-    // TODO: validate toUser.
-
-    switch (fields.type) {
-    case Message.ANNOUNCEMENT_TO_ALL_TYPE:
-    case Message.ANNOUNCEMENT_TO_NEW_TYPE:
-      if (!req.isAdmin) {
-        throw { status: 401 };
-      }
-      if (!fields.startDate) {
-        fields.startDate = new Date();
-      }
-      if (!fields.endDate) {
-        fields.endDate = new Date(fields.startDate.getTime() + 30*24*60*60*1000);
-      }
-    }
-
-    resolve(Message.create(fields));
-  }))
-});
-
-// Update message by ID.
-router.put("/:id", function(req, res) {
-  res.jsonResultOf(new Promise(function(resolve) {
-    var fields = VALIDATOR.prevalidateUpdate(req.body);
-    resolve(Message.findById(req.params.id)
-      .then(function(message) {
-        if (!message) {
-          throw { status: 404 };
-        }
-
-        fields = VALIDATOR.postvalidateUpdate(message, fields);
-        if (!fields) {
-          return message;
-        }
-
-        switch (fields.type) {
-        case Message.ANNOUNCEMENT_TO_ALL_TYPE:
-        case Message.ANNOUNCEMENT_TO_NEW_TYPE:
-          if (!req.isAdmin) {
-            throw { status: 401 };
-          }
-          break;
-        default:
-          if ("type" in fields) {
-            throw { body: { type: fields.type } };
-          }
-        }
-
-        return message.updateAttributes(fields);
-      })
-    );
-  }))
-});
-
-module.exports = router;
+/* messages_api.js */

```

Comment:
For some reason github is showing the entire contents of the two versions of the file side-by-side, instead of isolating the changes.  Any idea why?  This makes it difficult for me to find your changes.